File: MT/LMS.Core/ViewModel/TaskVM.cs

using LMS.Core.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS.Core.ViewModel
{
    public class UserTaskVM:BaseViewModel
    {
        public string? User { get; set; }
        public string? Module { get; set; }
        public string? Status { get; set; }
        public int StatusId { get; set; }
        public string? Priority { get; set; }
        public int PriorityId { get; set; }
        public string? UserId { get; set; }
        public int ModuleId { get; set; }
        public string? Title { get; set; }
        public string? UserPhoneNumber { get; set; }
        public string? DirectSupervisorPhoneNumber { get; set; }
        public float SP { get; set; }
        public string? Description { get; set; }
        public string? DirectSupervisorId { get; set; }
        public string? UserMail { get; set; }
        public string? DirectSupervisorEmail { get; set; }
        public string? DirectSupervisorName { get; set; }
        public List<AttachmentsDE> Attachments { get; set; }
        public int ClaimPercent { get; set; }
        public float RemainingSPs { get; set; }
        public UserTaskVM()
        {
            this.Attachments = new List<AttachmentsDE>();
        }
    }
}

File: Back Office/src/app/views/tms/models/task-vm.ts
export class TaskVM {
    directSupervisorEmail:string;
    directSupervisorId:string;
    directSupervisorName:string;
    directSupervisorPhoneNumber:string;
    userPhoneNumber:string;
    userMail:string;
    id: number;
    user:string;
    module:string;
    status:string;
    priority:string;
    priorityId:number= 1011001;
    title: string;
    sp: number;
    description: string;
    isActive: boolean = true;
    createdById: number =0
    createdOn: Date=new Date()
    modifiedById: number
    modifiedOn: Date=new Date()
    userId: string; 
    moduleId: number= 0
    statusId: number= 1010001
    freeSearch: string
    attachments?:AttachmentsVM[]= new Array()
    hasErrors: boolean
  
}
export class AttachmentsVM {
   // time:string;
    id: number;
    dBoperation:number;
    taskId:number;
    docPath: string;
    name:string;
    base64File:Text;
    size:string;
    isActive: boolean = true;
    createdById: number =0
    createdOn: Date=new Date()
    modifiedById: number
    modifiedOn: Date=new Date()
    // createdOn: Date=new Date()


}
